File: app/src/main/java/com/mai/nix/maiapp/MapFragment.java
package com.mai.nix.maiapp;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.design.widget.FloatingActionButton;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.webkit.WebView;

/**
 * Created by Nix on 13.05.2017.
 */

public class MapFragment extends Fragment {
    private WebView mWebView;
    private FloatingActionButton mActionButton;
    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View v = inflater.inflate(R.layout.map_fragment_layout, container, false);
        mWebView = (WebView) v.findViewById(R.id.campus_view);
        mActionButton = (FloatingActionButton)v.findViewById(R.id.legend_button);
        mWebView.getSettings().setSupportZoom(true);
        mWebView.getSettings().setBuiltInZoomControls(true);
        mWebView.setPadding(0, 0, 0, 0);
        mWebView.setScrollbarFadingEnabled(true);
        mWebView.setScrollBarStyle(View.SCROLLBARS_INSIDE_OVERLAY);
        mWebView.loadUrl("file:///android_res/drawable/scheme_for_app.png");
        mActionButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                LegendDialog dialog = new LegendDialog();
                dialog.show(getChildFragmentManager(), "kek");
            }
        });
        return v;
    }
}

File: app/src/main/java/com/mai/nix/maiapp/navigation_fragments/campus/MapFragment.kt
package com.mai.nix.maiapp.navigation_fragments.campus

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.mai.nix.maiapp.LegendDialog
import com.mai.nix.maiapp.R
import kotlinx.android.synthetic.main.fragment_map_layout.*

/**
 * Created by Nix on 13.05.2017.
 */

class MapFragment : Fragment() {
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_map_layout, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        campusView.settings.setSupportZoom(true)
        campusView.settings.builtInZoomControls = true
        campusView.setPadding(0, 0, 0, 0)
        campusView.isScrollbarFadingEnabled = true
        campusView.scrollBarStyle = View.SCROLLBARS_INSIDE_OVERLAY
        campusView.loadUrl("file:///android_asset/scheme_for_app.png")
        legendButton.setOnClickListener {
            val dialog = LegendDialog()
            dialog.show(childFragmentManager, "LegendDialog")
        }
    }
}